org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/JVMConfigurator.java

```diff
@@ -118,6 +118,10 @@ public static boolean configureJVMs(Preferences preferences, JavaClientConnectio
 						// https://github.com/eclipse/eclipse.jdt.ls/issues/1646
 						installType = JavaRuntime.getVMInstallType(MAC_OSX_VM_TYPE);
 					}
+					if (installType == null || installType.getVMInstalls().length == 0) {
+						// https://github.com/eclipse/eclipse.jdt.ls/issues/1696
+						continue;
+					}
```

Comment:
After this fix, launching jdt.ls with `jdt.ls.socket-stream.launch` launch file will skip `"configureJVMs"` job, that's not we want. The goal is just to use macOSX install type whenever possible. If that doesn't exist, then fall back to the standard install type. The change is something like below, and this can be a helper method for reuse.

```java
    IVMInstallType installType = JavaRuntime.getVMInstallType(StandardVMType.ID_STANDARD_VM_TYPE);
	if (installType == null || installType.getVMInstalls().length == 0) {
		// https://github.com/eclipse/eclipse.jdt.ls/issues/1646
        IVMInstallType macInstallType = JavaRuntime.getVMInstallType(MAC_OSX_VM_TYPE);
		if (macInstallType != null) {
			installType = macInstallType;
		}
	}
```

Also don't forget the similar logic in the method "configureDefaultVM".

Reply:
Fixed.